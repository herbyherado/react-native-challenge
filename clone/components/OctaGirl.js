import React, { Component } from 'react';
import { View, Text, Image, StyleSheet } from 'react-native'

class OctaGirl extends Component {
  render() {
    return (
      <View>
        <Image style={styles.img} source={{ uri: this.props.item.medium_profile_picture }}/>
        <Text>{`${this.props.item.first_name} ${this.props.item.last_name}`}</Text>
        <Text>{ this.props.item.country_residing }</Text>
        <Text>{ this.props.item.quote }</Text>
      </View>
    )
  }
};

const styles = StyleSheet.create({
  img: {
    width: 150,
    height: 150,
    alignSelf: 'center'
  }
})

export default OctaGirl