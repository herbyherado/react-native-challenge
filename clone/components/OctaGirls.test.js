import React, { Component } from 'react'
import enzyme, { shallow, mount } from 'enzyme'
import { Text, View, Image } from 'react-native'
import Adapter from 'enzyme-adapter-react-16'
import OctaGirl from './OctaGirl'
import renderer from 'react-test-renderer'

enzyme.configure({ adapter: new Adapter()})

let wrapper
let mockData

beforeEach(() => {
  mockData = {
    id: '6638',
    first_name: 'Arianny',
    last_name: 'Celeste',
    country_residing: 'USA',
    quote: 'The best part of the job is the whole experience.',
    medium_profile_picture: 'http://media.ufc.tv//fighter_images/Octagon_Girls/Arianny_MediumProfile.jpg',
  }

  wrapper = shallow(<OctaGirl item={ mockData }/>)
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<OctaGirl item={ mockData }/>)
    expect(tree).toMatchSnapshot()
  })
})

describe('<OctaGirl/>', () => {
  it('should render <OctaGirl/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('Image should be inserted', () => {
  it('should have source in <Image/>', () => {
    expect(wrapper.find('Image').filterWhere((item) => {
      return item.source === mockData.medium_profile_picture
    }))
  })
})

describe('Text props should return true', () => {
  it('should return text as expected', () => {
    expect(wrapper.find('Text').at(0).props().children).toEqual('Arianny Celeste')
    expect(wrapper.find('Text').at(1).props().children).toContain('USA')
    expect(wrapper.find('Text').at(2).props().children).toContain('The best part of the job is the whole experience.')
  })
})

describe('<OctaGirl /> child is rendered', () => {
  it('should render <OctaGirl/> child elements', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('View').length).toBe(1)
    expect(wrapper.find(<Text/>)).toBeDefined()
    expect(wrapper.find('Text').length).toBe(3)
    expect(wrapper.find(<Image/>)).toBeDefined()
    expect(wrapper.find('Image').length).toBe(1)
  })
})
