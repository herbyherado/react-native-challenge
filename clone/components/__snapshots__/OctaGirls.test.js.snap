// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View>
  <Image
    source={
      Object {
        "uri": "http://media.ufc.tv//fighter_images/Octagon_Girls/Arianny_MediumProfile.jpg",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 150,
        "width": 150,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Arianny Celeste
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    USA
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    The best part of the job is the whole experience.
  </Text>
</View>
`;
